### PR TITLE
feat(j-s): Indictment sent or withdrawn from prison notifications

### PR DESCRIPTION
**Pull request by unakb**
[Senda tilkynningu til FMST - fullnusta](https://app.asana.com/0/1199153462262248/1208708538371121)
[Senda tilkynningu til FMST - afturköllun ](https://app.asana.com/0/1199153462262248/1208708538371122)

## What

Send notifications to a group of prison admin staff when an indictment is sent or withdrawn .

## Why

To let them know that something new has happened in case they need to react to it. 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced new notification types for indictments related to prison administration.
  - Added functionality to send notifications when a defendant is sent to or withdrawn from prison administration.
  - New secret added for managing prison administration indictment emails.

- **Bug Fixes**
  - Improved handling of notification messages for defendant status changes.

- **Documentation**
  - Updated configuration files to include new secret for prison administration email notifications.

- **Tests**
  - Added unit tests for new notification functionalities related to indictments.

- **Chores**
  - Restructured import paths for better organization within the notification services.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/defendant/defendant.service.ts [hunk 1]
```diff
@@ -171,9 +191,20 @@ export class DefendantService {
...
         }
       }
+    } else if (
+      updatedDefendant.isSentToPrisonAdmin !== undefined &&
+      updatedDefendant.isSentToPrisonAdmin !== oldDefendant.isSentToPrisonAdmin
+    ) {
+      messages.push(
+        this.getMessagesForIndictmentToPrisonAdminChanges(updatedDefendant),
+      )
+    }
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/defendant/defendant.service.ts:201]: The field does have those decorators as far as I can tell, can you double check this comment?

**Review by peturgq — APPROVED** [review]

File: apps/judicial-system/backend/src/app/modules/notification/internalNotification.controller.ts [hunk 2]
```diff
@@ -34,12 +34,12 @@ import { InstitutionNotificationDto } from './dto/institutionNotification.dto'
 import { NotificationDispatchDto } from './dto/notificationDispatch.dto'
 import { SubpoenaNotificationDto } from './dto/subpoenaNotification.dto'
 import { DeliverResponse } from './models/deliver.response'
-import { CaseNotificationService } from './caseNotification.service'
-import { CivilClaimantNotificationService } from './civilClaimantNotification.service'
-import { DefendantNotificationService } from './defendantNotification.service'
-import { InstitutionNotificationService } from './institutionNotification.service'
+import { CaseNotificationService } from './services/caseNotification/caseNotification.service'
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/notification/internalNotification.controller.ts:37]: Nice, time to move to subfolders.

File: libs/judicial-system/types/src/lib/notification.ts [hunk 3]
```diff
@@ -64,6 +66,8 @@ export enum NotificationType {
   SERVICE_FAILED = SubpoenaNotificationType.SERVICE_FAILED,
   SPOKESPERSON_ASSIGNED = CivilClaimantNotificationType.SPOKESPERSON_ASSIGNED,
   INDICTMENTS_WAITING_FOR_CONFIRMATION = InstitutionNotificationType.INDICTMENTS_WAITING_FOR_CONFIRMATION,
+  INDICTMENT_SENT_TO_PRISON_ADMIN = DefendantNotificationType.INDICTMENT_SENT_TO_PRISON_ADMIN,
```
Comment (gudjong) [inline libs/judicial-system/types/src/lib/notification.ts:69]: For easy reading please consider grouping these with other defendant notifications.

**Review by gudjong — APPROVED** [review]
Realy nice. One suggestion.

**Pull request merged**